### ISSUE TITLE
[k3s-upgrade] not working for K3OS installe K3S

### ISSUE THREAD

**Issue by bkraegelin** [k3s-io/k3s#5047]
**Environmental Info:**
K3s Version: 

k3s version v1.21.1+k3s1 (75dba57f)
go version go1.16.4

Node(s) CPU architecture, OS, and Version: 

Linux k3s11 5.4.0-73-generic #82 SMP Thu Jun 3 02:29:43 UTC 2021 x86_64 GNU/Linux

Cluster Configuration:

1 Server, 2 Agents

**Describe the bug:**

Error Message: sha256sum: can't open '/host/sbin/k3s': No such file or directory

**Steps To Reproduce:**

- Installed K3s: 
K3OS ISO image v0.21.1-k3s1r0

**Expected behavior:**

Upgrade K3S to next version.

**Actual behavior:**

Job ends normally without any upgrades.

**Additional context / logs:**

/sbin/k3s is symlink to /k3os/system/k3s/current/k3s

Same issue as https://github.com/rancher/system-upgrade-controller/issues/90
filed as issue https://github.com/rancher/system-upgrade-controller/issues/178

**Backporting**

- [ ] Needs backporting to older releases

**Comment by brandond** [comment]
Please confirm that you're using the k3os-specific upgrade plans from https://github.com/rancher/k3os#automatic-upgrades - not the example plans from the system-upgrade-controller repo.

**Comment by bkraegelin** [comment]
Thanks for your comment. Indeed, my problem arises from the K3S example plan.

I started with system-upgrade-controller and the examples for K3S and failed. So I postponed upgrading K3OS until recently, when I found the issue https://github.com/rancher/system-upgrade-controller/issues/90. I verified that on K3OS, which also uses symlinks, which is not covered by the K3S upgrade procedure.

In the meantime I successfully upgraded K3OS using the specific upgrade plan which also upgrades K3S. I didn't realize that before. So my problem is solved.

But I like to suggest enhancing https://github.com/k3s-io/k3s-upgrade/blob/master/scripts/upgrade.sh, function replace_binary(), with something like
`if [ -L $FULL_BIN_PATH] ; then`
``K3S_BIN_PATH = `readlink $FULL_BIN_PATH` ``
`FULL_BIN_PATH = "/host$K3S_BIN_PATH"`
`fi`

That may help others. Thanks for your help,
Birger
